src/screens/Main/Main.screen.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {FC, useEffect} from 'react';
+import {Text, ScrollView} from 'react-native';
+import {useSelector, useDispatch} from 'react-redux';
+import {publicChannelsSelectors} from '../../store/publicChannels/publicChannels.selectors';
+import {publicChannelsActions} from '../../store/publicChannels/publicChannels.slice';
+import {IChannelInfo} from '../../store/publicChannels/publicChannels.types';
+
+export const MainScreen: FC = () => {
+  const channels = useSelector(publicChannelsSelectors.publicChannels);
+  const messages = useSelector(publicChannelsSelectors.currentChannelMessages);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(publicChannelsActions.getPublicChannels());
```

Comment:
I think you fetch public channels from saga, so no need for doing it here

Reply:
Not exactly. There is a need to wait for event when storage is initialized. Otherwise, if we request them too early, they wont be returned and we won't subscribe for their event. Waggle itself casts event telling us that storage has been initialized so there is no saga that could be used to automatically request the channels. On ZbayLite it's handled by Electron's callbacks.

Reply:
imo the saga should just wait in a loop, try to fetch and `yield delay(...)` in each iteration

Reply:
if you have such trigger logic in a certain component, it will later become quite complex when you have multiple screens and some of them co-exist (offscreen)

It's much better to drive all interaction with waggle from saga, because it's under much stricter control